### PR TITLE
Resolve script parsing error when changing from bash to sh.

### PR DESCRIPTION
**Pull request by lostluck**
Resolve script parsing error when changing from bash to sh.

https://unix.stackexchange.com/questions/45781/shell-script-fails-syntax-error-unexpected

Basically, unless "bash" is specifically invoked in ubuntu flavours of linux, it uses it's "dash" shell instead, which has very basic features.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @kileys

**Comment by lostluck** [comment]
I'll need to cherry pick this into the release branch too. My local flavour of linux appears to be very opinionated on which script tool is parsing the scripts.

File: release/src/main/scripts/set_version.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 set -e
 
-function usage() {
+function usage(){
```
Comment (kileys) [inline release/src/main/scripts/set_version.sh:27]: Does this cause a problem?

Comment (lostluck) [reply release/src/main/scripts/set_version.sh:27]: No, removing. This is just where the "dash" interpreter dies. Thanks!

**Review by kileys — APPROVED** [review]

**Pull request merged**